src/base/method/baseContainer.ts

```diff
@@ -0,0 +1,45 @@
+import { BaseMethod, BaseMethodResponse } from './baseMethod';
+import { MethodABI } from '../iface';
+
+/**
+ * Wrapper class for methods with the same name
+ * Since methods can overload each other, we need this to be able to call methods by name and
+ *   intelligently determine the correct one to call by parameter
+ */
+export abstract class BaseMethodContainer<K extends MethodABI> {
```

Comment:
Instead of abstract class with required implementation of call, I would just make this a single standalone class `MethodContainer` and add a new interface for the `call` like `MethodCallBuilder`. implement `MethodCallBuilder` for each coin. Have `MethodContainer` take a `MethodCallBuilder` instance as constructor parameter and use it to build the call coin-generically. 
More composable, flexible solution rather than rather rigid solution forced by `abstract class`es

Reply:
yeah I think I agree here. we will probably have to move towards one abstract class in the future that could instantiate  `K`'s with multiple interfaces as inheritance. for instance, Tron is solidity based but also diverges a tiny bit so its ok within this system. however we would want to describe these methods in other non-solidity languages, I'd rather `MethodABI` be more flexible than just one way to implement around solidity. in theory, we could be using any other smart contract language in the future that has its own calls and we would need to reason about that surface as well